lib/pubsub/topic.js

```diff
@@ -63,6 +63,27 @@ function Topic(pubsub, options) {
 }
 
 /**
+ * Format a message object as the upstream API expects it.
+ *
+ * @private
+ *
+ * @return {object}
+ */
+Topic.formatMessage_ = function(message) {
+  if (!message.data) {
+    message = {
+      data: new Buffer(message).toString('base64')
+    };
+  }
+
+  if (!util.is(message.data, 'string') && !util.is(message.data, 'buffer')) {
+    message.data = JSON.stringify(message.data);
```

Comment:
> ``` js
// multiple
topic.publish([
 { data: "anyType", labels: [] },
 { data: "anyType", labels: [] }
], callback)

> // single
topic.publish({ data: "anyType", labels: [] }, callback)
```

Seems fine to me. And it's understandable. We could change `data:` to `message:` to make it a little friendlier. We do a similar thing with datastore already so I don't see a problem. And very easy to extend in the future! :)